### ISSUE TITLE
Allowing sections/options to be copied from one document to the other

### ISSUE THREAD

**Issue by abravalheri** [pyscaffold/configupdater#47]
Right now it is not possible to simply copy a section or option object from one container to another.
When we try to add the copied object to the other container a `AlreadyAttechedError` is raised.

This is a symptom of the fact that every block contains a reference to their containers and when they are copied, the copy reference keeps pointing to the previous container.

I think copies should always be created in a "detached" mode, i.e., the reference to the parent container should be `None`.
To solve this problem we also have to rely on `deepcopy`, because when copying a section, the references inside its options should also be clear (in my understanding it makes very little sense to implement shallow copies in ConfigUpdater).

**Comment by abravalheri** [comment]
Hi @FlorianWilhelm, I provided a series of PRs to address this issue. I am using a technique called [stacked pull requests](https://www.michaelagreiler.com/stacked-pull-requests/) to facilitate the review, which means that merging of the associated PRs in the reverse order makes things easier 😝.

In each PR I add a little bit of description/discussion about what is done.

Last time we talked I mentioned that stdlib's `Mapping`  could be providing a dict-like `copy` method, but after investigation I don't see that being the case ([cpython's implementation](https://github.com/python/cpython/blob/3.9/Lib/_collections_abc.py) don't define copy for either Mapping or MutableMapping). So I just decided to not implement a `copy` method.
